docs/guides/gitlab.md

```diff
@@ -0,0 +1,280 @@
+:::
+
+### Prepare The Environment
+
+1.  Create volumes for persistent data. This guide uses the base directory `/srv/gitlab`; adjust this path for your local environment:
+
+    ```bash
+    mkdir -p /srv/gitlab #Adjust the path based on your common Docker volume location.
+    ```
+
+1.  Create three sub-directories: `config`, `data`, and `logs`:
+
+    ```bash
+    mkdir -p /srv/gitlab/{config,data,logs}
+    ```
+
+### Install and Configure GitLab
+
+1.  Edit your `docker-compose.yml` file to define a new service for GitLab:
+
+    ```yaml
+    services:
+
+    ...
+
+    gitlab:
+      container_name: gitlab
+      image: gitlab/gitlab-ee:latest
+      environment:
+        GITLAB_OMNIBUS_CONFIG: |
+          external_url 'https://gitlab.pomerium.localhost.io'
+          registry_external_url 'https://gitlab.pomerium.localhost.io'
+          letsencrypt['enable'] = false
+          nginx['listen_port'] = 80
+          nginx['listen_https'] = false
+          registry_nginx['listen_port'] = 80
+          registry_nginx['listen_https'] = false
+          pages_nginx['listen_port'] = 80
+          pages_nginx['listen_https'] = false
+          mattermost_nginx['listen_port'] = 80
+          mattermost_nginx['listen_https'] = false
+      volumes:
+        - '/srv/gitlab/config:/etc/gitlab'
+        - '/srv/gitlab/logs:/var/log/gitlab'
+        - '/srv/gitlab/data:/var/opt/gitlab'
+      expose:
+        - 80
+        - 443
+        - 22
+      restart: always
+      shm_size: '256m'
+    ```
+
+    - Adjust `external_url` and `registry_external_url` to match the external path, which we will define in Pomerium later in the process.
+    - Adjust the paths under `volumes` to match the directories created in [the previous section](#prepare-the-environment).
+
+1.  Bring up the new Docker Compose configuration:
+
+    ```bash
+    docker-compose up -d
+    ```
+
+    The container may take several minutes to initialize. Once complete, the status provided to `docker ps` will change from `health:starting` to `healthy`.
+    
+    You can also monitor the progress with `docker logs -f gitlab`. Note that even after the process is complete, the log file will continue to output log messages.
+
+## Configure a Pomerium Route
+
+Edit `config.yaml` and add a route for GitLab. Note that this example assumes access to the mkcert CA certificate at `/mkcert/rootCA.pem`:
+
+```yaml
+  - from: https://gitlab.localhost.pomerium.io
+    to: http://gitlab
+    preserve_host_header: true
+    policy:
+      - allow:
+          or:
+            - domain:
+                is: example.com
+```
+
+Once the route is applied, you should be able to access GitLab from `https://gitlab.localhost.pomerium.io`. Note that when using Docker, you may need to restart the Pomerium container to apply the changes.
```

Comment:
```suggestion
Once the route is applied, you should be able to access GitLab from `https://gitlab.localhost.pomerium.io`. Note that when using Docker, you may need to restart the Pomerium container to apply the changes as file change detection can be finicky on mounted docker volumes.
```

Not sure about verbage but I just want to point out this is a docker thing, not a us thing. Our hot reload code works but docker is weird.